src/TestSuite/EmailAssertTrait.php

```diff
@@ -0,0 +1,153 @@
+     * @return \Cake\Mailer\Mailer|\PHPUnit_Framework_MockObject_MockObject
+     */
+    public function getMockForMailer($className, array $methods = [])
+    {
+        $name = current(array_slice(explode('\\', $className), -1));
+
+        if (!in_array('profile', $methods)) {
+            $methods[] = 'profile';
+        }
+
+        $mailer = $this->getMockBuilder($className)
+            ->setMockClassName($name)
+            ->setMethods($methods)
+            ->setConstructorArgs([$this->email()])
+            ->getMock();
+
+        $mailer->expects($this->any())
+            ->method('profile')
+            ->willReturn($mailer);
+
+        return $mailer;
+    }
+
+    /**
+     * @param string $needle Text to look for.
+     * @param string $message The failure message to define.
+     * @return void
+     */
+    public function assertEmailMessageContains($needle, $message = null)
+    {
+        $this->assertEmailHtmlMessageContains($needle, $message);
+        $this->assertEmailTextMessageContains($needle, $message);
+    }
+
+    /**
+     * @param string $needle Text to look for.
+     * @param string $message The failure message to define.
+     * @return void
+     */
+    public function assertEmailHtmlMessageContains($needle, $message = null)
+    {
+        $haystack = $this->email()->message('html');
+        $this->assertTextContains($needle, $haystack, $message);
+    }
+
+    /**
+     * @param string $needle Text to look for.
+     * @param string $message The failure message to define.
+     * @return void
+     */
+    public function assertEmailTextMessageContains($needle, $message = null)
+    {
+        $haystack = $this->email()->message('text');
+        $this->assertTextContains($needle, $haystack, $message);
+    }
+
+    /**
+     * @param string $expected Email's subject.
+     * @param string $message The failure message to define.
+     * @return void
+     */
+    public function assertEmailSubject($expected, $message = null)
+    {
+        $result = $this->email()->subject();
+        $this->assertSame($expected, $result, $message);
+    }
+
+    /**
+     * @param string $email Sender's email address.
+     * @param string $name Sender's name.
+     * @param string $message The failure message to define.
+     * @return void
+     */
+    public function assertEmailFrom($email, $name, $message = null)
+    {
+        $expected = [$email => $name];
+        $result = $this->email()->from();
+        $this->assertSame($expected, $result, $message);
+    }
+
+    /**
+     * @param string $email Sender's email address.
+     * @param string $name Sender's name.
+     * @param string $message The failure message to define.
+     * @return void
+     */
+    public function assertEmailTo($email, $name, $message = null)
+    {
+        $expected = [$email => $name];
+        $result = $this->email()->to();
+        $this->assertSame($expected, $result, $message);
+    }
```

Comment:
and also maybe a `assertEmailToContains()` to look for an email address in a list